docs/src/util.ts

```diff
@@ -1,15 +1,23 @@
 export function getLines(str: string, lineStart: number, lineEnd?: number): string {
-    return str.split('\n').slice(lineStart - 1, lineEnd || lineStart).join('\n');
+    let lines = str.split('\n').slice(lineStart - 1, lineEnd || lineStart);
+    const leadingWhitespace = Math.min(...lines.map(line => line.search(/\S/)).map(Number).filter(n => 0 !== n));
+    if (0 < leadingWhitespace) {
+        lines = lines.map(line => line.slice(leadingWhitespace));
+    }
+    return lines.join('\n');
```

Comment:
Minor: can you leave a 1 line comment about what this is doing?

Reply:
Sure